examples/video_player/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+use eframe::egui::{self, Grid};
+use eframe::egui::{CentralPanel, TextEdit};
+use humanize_duration::{prelude::*, Truncate};
+use kornia::image::{Image, ImageSize};
+use kornia::imgproc::resize::resize_fast;
+use kornia::io::stream::video::{ImageFormat, SeekFlags, State, VideoReader};
+use std::path::PathBuf;
+
+const SLIDER_HEIGHT: f32 = 20.;
+
+pub struct MyApp {
+    file_path: String,
+    video_reader: Option<VideoReader>,
+    slider_pos: u64,
+    playback_speed: u32,
+}
+
+impl Default for MyApp {
+    fn default() -> Self {
+        Self {
+            file_path: String::new(),
+            video_reader: None,
+            slider_pos: 0,
+            playback_speed: 1,
+        }
+    }
+}
+
+impl eframe::App for MyApp {
+    fn update(&mut self, ctx: &eframe::egui::Context, _frame: &mut eframe::Frame) {
+        ctx.request_repaint();
+        CentralPanel::default().show(ctx, |ui| {
+            ui.horizontal(|ui| {
+                ui.add_enabled_ui(self.video_reader.is_some(), |ui| {
+                    if ui.button("clear").clicked() {
+                        self.video_reader = None;
+                    }
+                });
+
+                ui.add_enabled_ui(self.video_reader.is_none(), |ui| {
+                    if ui.button("load").clicked() {
+                        log::info!("Loading the Video");
+                        let file_path = PathBuf::from(&self.file_path);
+                        log::info!("File Path: {}", &self.file_path);
+                        let mut video_reader = match VideoReader::new(&file_path, ImageFormat::Rgb8)
+                        {
+                            Ok(v) => v,
+                            Err(err) => {
+                                log::error!("Failed to create VideoReader: {}", err);
+                                return;
+                            }
+                        };
+
+                        // Start the video
+                        match video_reader.start() {
+                            Ok(_) => {}
+                            Err(err) => {
+                                log::error!("Failed to start the video: {}", err);
+                                return;
+                            }
+                        };
+                        self.video_reader = Some(video_reader);
+                    }
+
+                    ui.add_sized(
+                        [ui.available_width(), ui.available_height()],
+                        TextEdit::singleline(&mut self.file_path).hint_text("Click to set path"),
+                    );
+                });
+            });
+
+            if let Some(video_reader) = &mut self.video_reader {
```

Comment:
maybe to avoid to avoid too much verbosity, just `.unwrap` with a `// SAFETY` comment